x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -258,6 +260,15 @@ export const createLifecycleExecutor = (logger: Logger, ruleDataClient: RuleData
     event[ALERT_START] = started;
     event[ALERT_UUID] = alertUuid;
 
+    // not sure why typescript needs the non-null assertion here
+    // we already assert the value is not undefined with the ternary
+    // still getting an error with the ternary.. strange.
```

Comment:
This is a Typescript limitation. It is not smart enough to narrow down the type of a computed property inside if-else:

```ts
const obj: { prop?: string } = {}

if (obj.prop) {
  obj.prop.toLowerCase() // <- guard works, no type error
}

const prop: 'prop' = 'prop'
if (obj[prop]) {
  obj[prop].toLowerCase() // <- guard doesn't work, object is possibly 'undefined' error
}


```